### ISSUE TITLE
[Improvement] Support combine operation in map stage.

### ISSUE THREAD

**Issue by zhengchenyu** [apache/incubator-uniffle#1238]
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-uniffle/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

When map send buffer to shuffle server,  we could reduce data transmission as much as possible through combine operations.

Because most data may be aggregated in the shuffle server, only part of the data can be combined on the map side. , so the proportion of merge may not be very high.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

**Comment by zuston** [comment]
Good idea. The combine could be supported on the different computing frameworks, like Spark/Tez/MR , right?
